packages/nimble-components/src/table-column/number-text/models/integer-formatter.ts

```diff
@@ -0,0 +1,13 @@
+/**
+ * The formatter for a number-text column whose format is configured to be 'integer'.
+ */
+export class IntegerFormatter {
+    private static readonly formatter = new Intl.NumberFormat(undefined, {
+        maximumFractionDigits: 0,
+        useGrouping: true
+    });
+
```

Comment:
Chatted with @atmgrifter00 a bit about the following examples:
![image](https://github.com/ni/nimble/assets/1588923/875b3943-3f40-420a-953e-30aee2ab9205)


The concerns were the following:

- Us ever showing negative zero in the table for our pre-made configs (default and integer) is not a great built-in behavior. Negative zero vs zero distinctions are really only relevant for specific engineering scenarios and shouldn't be the default representation.
- The name "integer" does not reflect what is happening well from a programmer configuration expectation. "Normal" / programmer expected float to integer conversion should not have negative zeros (because that can't be represented in the expected two's complement integer format) and should truncate instead of round.

I think the following changes would address those concerns:

- We should rename the `integer` format to `round`. Seems to better represent what is described by the docs and makes it clearer what will happen to your numbers.
- We should use the `signDisplay: "negative"` configuration for all these modes. i.e. the `round` mode and in both the formatters used in the `default` mode.